src/components/hoc/MaybeCurrencySetting.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import { type Cleaner, asMaybe, uncleaner } from 'cleaners'
+import { type EdgeCurrencyConfig } from 'edge-core-js'
+import * as React from 'react'
+
+import { useWatchCurrencyConfig } from '../../hooks/useWatch.js'
+import { useCallback, useMemo } from '../../types/reactHooks.js'
+
+/**
+ * Specific settings sections will receive these cleaned props.
+ */
+export type CurrencySettingProps<T, X = void> = {|
+  // The starting value, taken from the plugin's currency info:
+  defaultSetting: T,
+
+  // The current value, taken from disk:
+  setting: T,
+
+  // Extra information passed to the component at creation time,
+  // to customize its appearance or such:
+  extraInfo: X,
+
+  // Called when the user tries to update the setting:
+  onUpdate(setting: T): Promise<void>
+|}
+
+type Props = {|
+  currencyConfig: EdgeCurrencyConfig
+|}
+
+/**
+ * Renders a settings section, but only if the cleaner passes.
+ */
+export function maybeCurrencySetting<T, X>(
+  Component: React$StatelessFunctionalComponent<CurrencySettingProps<T, X>>,
```

Comment:
Where does this type come from btw `React$StatelessFunctionalComponent`?

Reply:
It is built into Flow. It's a global, always-on symbol, just like `Array` or `$Call` are built in.